scheduler/mesos/mesos.go

```diff
@@ -0,0 +1,88 @@
+package mesos
+
+import (
+	"errors"
+	"sync"
+
+	"github.com/docker/swarm/cluster"
+	"github.com/docker/swarm/discovery"
+	"github.com/docker/swarm/scheduler/options"
+	"github.com/samalba/dockerclient"
+)
+
+var ErrNotImplemented = errors.New("not implemented in the mesos scheduler")
+
+type MesosScheduler struct {
+	sync.Mutex
+
+	//TODO: list of mesos masters
+	cluster *cluster.Cluster
+	options *options.SchedulerOptions
+}
+
+func (s *MesosScheduler) Initialize(options *options.SchedulerOptions) {
+	s.options = options
+
+	s.cluster = cluster.NewCluster(s.options.Store)
+}
+
+// Schedule a brand new container into the cluster.
+func (s *MesosScheduler) CreateContainer(config *dockerclient.ContainerConfig, name string) (*cluster.Container, error) {
+
+	s.Lock()
+	defer s.Unlock()
+
+	//TODO: RequestOffers from mesos master
+
+	//TODO: pick the right offer (using strategy & filters ???)
+
+	//TODO: LaunchTask on the Mesos cluster and get container
+
+	//TODO: Store container in store
+	//  -   s.cluster.CommitContainerInStore(container.Id, config, name)
+
+	return nil, ErrNotImplemented
+}
+
+// Remove a container from the cluster. Containers should always be destroyed
+// through the scheduler to guarantee atomicity.
+func (s *MesosScheduler) RemoveContainer(container *cluster.Container, force bool) error {
+	s.Lock()
+	defer s.Unlock()
+
+	//TODO: KillTask
+
+	//TODO: remove container
+	//  -   s.cluster.RemoveContainerFromStore(container.Id, force)
+
+	return ErrNotImplemented
+}
+
+// Entries are Mesos masters
+func (s *MesosScheduler) NewEntries(entries []*discovery.Entry) {
```

Comment:
I'm not sure I understand what this function is for.
The comment above says entries are mesos masters, but in the code it's adding to the cluster which I assume is slave nodes with swarm launched.
For Mesos there isn't any need to look for new nodes, since when new nodes are added to the data center the framework will get offers from it from the resourceOffer callback, and since you launched the task yourself you also know what slave that is, in the framework side just need to listen on the statusUpdate on TASK_RUNNING to get the information.